internal/encryption/keyprovider/openbao/mock_test.go

```diff
@@ -0,0 +1,19 @@
+package openbao
+
+import (
+	"context"
+
+	openbao "github.com/openbao/openbao/api"
+)
+
+type mockClientFunc func(ctx context.Context, path string, data []byte) (*openbao.Secret, error)
+
+func (f mockClientFunc) WriteBytesWithContext(ctx context.Context, path string, data []byte) (*openbao.Secret, error) {
+	return f(ctx, path, data)
+}
+
+func injectMock(m mockClientFunc) {
+	newClient = func(_ *openbao.Config, _ string) (client, error) {
```

Comment:
Where does this get cleaned up? A `t.Cleanup` function would be in order here.

Reply:
Makes sense. I added a cleanup here: [7021c8d](https://github.com/opentofu/opentofu/pull/1436/commits/7021c8d304c5def03b23d6f9d4f29718e926e4f6).